crates/bevy_ecs/src/system/into_system.rs

```diff
@@ -15,6 +16,7 @@ pub struct SystemState {
     pub(crate) archetype_component_access: Access<ArchetypeComponentId>,
     // NOTE: this must be kept private. making a SystemState non-send is irreversible to prevent SystemParams from overriding each other
     is_send: bool,
+    pub(crate) system_counter: Option<u32>,
```

Comment:
The system counters being optional is just a crutch during initial development to make sure everything is correctly initialized.